azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -892,10 +893,27 @@ private void insertJVMAargs() {
     final String previousJVMArgs = this.props.get(JavaProcessJob.JVM_PARAMS);
     jobJVMArgs += (previousJVMArgs == null) ? "" : " " + previousJVMArgs;
 
+    // Add useful Java options for java jobs which are provided through properties
+    final String javaOpts = insertJavaOptions();
+    jobJVMArgs += (javaOpts == null) ? "" : " " + javaOpts;
     this.logger.info("job JVM args: " + jobJVMArgs);
     this.props.put(JavaProcessJob.JVM_PARAMS, jobJVMArgs);
   }
 
+  /**
+   * Add useful Java options for java jobs provided through properties.
+   */
+  private String insertJavaOptions() {
```

Comment:
I see no reason not to have the method be final if we are having the calling variable be final, but seeing as we intend to delete this, that's a-ok.